--- kbd-1.12/src/Makefile.in.orig	2004-01-03 17:53:54.000000000 +0000
+++ kbd-1.12/src/Makefile.in	2004-08-01 04:49:04.602780040 +0000
@@ -46,8 +46,9 @@
 OLDPROGS= mapscrn loadunimap
 
 # Not installed by default
-MISC    = screendump setlogcons setvesablank spawn_console spawn_login \
-	  getunimap clrunimap outpsfheader setpalette
+#MISC    = screendump setlogcons setvesablank spawn_console spawn_login \
+#	  getunimap clrunimap outpsfheader setpalette
+MISC     = setlogcons setvesablank getunimap
 
 # Installed by default
 SHCMDS  = unicode_start unicode_stop
@@ -73,7 +74,7 @@
 install:	all
 	install -d -m 0755 $(BINDIR) $(LOADKEYS_BINDIR)
 	install -s -m 0755 $(setowner) $(PROGS) $(OLDPROGS) $(BINDIR)
-#	install -s -m 0755 $(setowner) $(MISC) $(BINDIR)
+	install -s -m 0755 $(setowner) $(MISC) $(BINDIR)
 	install -c -m 0755 $(setowner) $(SHCMDS) $(BINDIR)
 	for i in psfaddtable psfgettable psfstriptable; do \
 		rm -f $(BINDIR)/$$i; ln -s psfxtable $(BINDIR)/$$i; \
