diff -uNr util-linux-2.12q-orig/fdisk/sfdisk.c util-linux-2.12q/fdisk/sfdisk.c
--- util-linux-2.12q-orig/fdisk/sfdisk.c	2005-01-05 09:31:57.000000000 +1100
+++ util-linux-2.12q/fdisk/sfdisk.c	2005-04-04 23:01:08.000000000 +1000
@@ -172,7 +172,7 @@
  * Note: we use 512-byte sectors here, irrespective of the hardware ss.
  */
 #undef use_lseek
-#if defined (__alpha__) || defined (__ia64__) || defined (__x86_64__) || defined (__s390x__)
+#if defined (__alpha__) || defined (__ia64__) || defined (__x86_64__) || defined (__s390x__) || defined(__mips64)
 #define use_lseek
 #endif
 
