diff -ur mas-cvs-20050207/clients/util/mas-config.cpp mas-cvs-mod/clients/util/mas-config.cpp
--- mas-cvs-20050207/clients/util/mas-config.cpp	2004-09-03 08:26:36.000000000 +1000
+++ mas-cvs-amd64/clients/util/mas-config.cpp	2005-02-10 00:19:20.325851896 +1100
@@ -1,9 +1,9 @@
-#!/bin/sh
+XCOMM!/bin/sh
 
-#
-# mas-config.cpp - builds "mas-config" that can echo libraries,
-#                  includes, and defines ala gtk's gtk-config.
-#
+XCOMM
+XCOMM mas-config.cpp - builds "mas-config" that can echo libraries,
+XCOMM                  includes, and defines ala gtk's gtk-config.
+XCOMM
 
 show_usage()
 {
diff -ur mas-cvs-20050207/clients/util/mas-launch.cpp mas-cvs-mod/clients/util/mas-launch.cpp
--- mas-cvs-20050207/clients/util/mas-launch.cpp	2004-06-15 07:32:10.000000000 +1000
+++ mas-cvs-amd64/clients/util/mas-launch.cpp	2005-02-10 00:20:13.329794072 +1100
@@ -1,6 +1,6 @@
-#!/bin/bash
+XCOMM!/bin/bash
 
-# kill esound if it's running -- we can't coexist.
+XCOMM kill esound if it's running -- we can't coexist.
 pkill esd
 
 if [[ $1 == "-t" ]];
diff -ur mas-cvs-20050207/common/mas_endian.h mas-cvs-mod/common/mas_endian.h
--- mas-cvs-20050207/common/mas_endian.h	2003-01-23 00:26:17.000000000 +1100
+++ mas-cvs-amd64/common/mas_endian.h	2005-02-07 23:52:27.000000000 +1100
@@ -137,6 +137,8 @@
 #if (!defined(MAS_BIG_ENDIAN) && !defined(MAS_LITTLE_ENDIAN))
 
 #if	defined(__i386__) || defined(__ia64__) || defined(WIN32) || \
+	defined(__amd64__) || defined(__x86_64__) || \
+	defined(amd64__) || defined(x86_64__) || \
 	(defined(__alpha__) || defined(__alpha)) || \
 	defined(__arm__) || \
 	(defined(__mips__) && defined(__MIPSEL__)) || \
diff -ur mas-cvs-20050207/config/Imake.cf mas-cvs-mod/config/Imake.cf
--- mas-cvs-20050207/config/Imake.cf	2003-03-11 08:45:28.000000000 +1100
+++ mas-cvs-amd64/config/Imake.cf	2005-02-08 00:42:27.000000000 +1100
@@ -401,17 +401,21 @@
 # ifdef i386
 #  define i386Architecture
 #  undef i386
+# endif
 # ifdef __i386__
 #  ifndef i386Architecture
 #   define i386Architecture
 #  endif
 #  undef __i386__
 # endif /* i386 */
+# ifdef __x86_64__
+#  undef __x86_64__
+#  define AMD64Architecture
+# endif /* x86_64 */
 # ifdef __alpha
 #  define AlphaArchitecture
 #  undef __alpha
 # endif /* __alpha */
-#endif
 #endif /* linux */
 
 #ifdef __uxp__
diff -ur mas-cvs-20050207/config/Imake.tmpl mas-cvs-mod/config/Imake.tmpl
--- mas-cvs-20050207/config/Imake.tmpl	2003-12-11 10:21:08.000000000 +1100
+++ mas-cvs-amd64/config/Imake.tmpl	2005-02-09 21:56:46.703200192 +1100
@@ -337,14 +337,18 @@
 # define ExportDefines
 #endif
 
+#ifndef LibDirName
+# define LibDirName lib
+#endif
+
 #ifndef UsrLibDir
 #ifdef ProjectRoot
-#define UsrLibDir Concat(ProjectRoot,/lib)
+#define UsrLibDir Concat(ProjectRoot,/LibDirName)
 #ifndef AlternateUsrLibDir
 #define AlternateUsrLibDir YES
 #endif
 #else
-#define UsrLibDir /usr/lib
+#define UsrLibDir Concat(/usr,/LibDirName)
 #ifndef AlternateUsrLibDir
 #define AlternateUsrLibDir NO
 #endif
@@ -597,7 +601,7 @@
 #define GzipCmd gzip
 #endif
 #ifndef CppCmd
-#define CppCmd /lib/cpp
+#define CppCmd /LibDirName/cpp
 #endif
 #ifndef CppNoLineInfoOption
 #define CppNoLineInfoOption /**/
diff -ur mas-cvs-20050207/config/linux.cf mas-cvs-mod/config/linux.cf
--- mas-cvs-20050207/config/linux.cf	2003-03-11 08:45:28.000000000 +1100
+++ mas-cvs-amd64/config/linux.cf	2005-02-09 22:01:41.425395608 +1100
@@ -195,6 +195,17 @@
 #endif
 #endif /* i386Architecture */
 
+#ifdef AMD64Architecture
+#define OptimizedCDebugFlags	DefaultGcc2AMD64Opt
+#define LinuxMachineDefines	-D__x86_64__
+#define ServerOSDefines		XFree86ServerOSDefines -DDDXTIME
+#define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+#if UseElfFormat
+#define HasPlugin		YES
+#define VendorHasX11R6_3libXext	YES	/* XC or XFree86 3.3.1 */
+#endif
+#endif /* AMD64Architecture */
+
 #ifdef AlphaArchitecture
 #define OptimizedCDebugFlags	-O2
 #define LinuxMachineDefines	-D__alpha__
@@ -210,6 +221,20 @@
 #define ServerExtraDefines      -DGCCUSESGAS XFree86ServerDefines
 #endif /* Mc68020Architecture */
 
+#ifndef HaveLib64
+# if defined (AMD64Architecture)
+#  define HaveLib64     YES
+# else
+#  define HaveLib64     NO
+# endif
+#endif
+
+#if HaveLib64
+# ifndef LibDirName
+#  define LibDirName            lib64
+# endif
+#endif
+
 #define StandardDefines		-Dlinux LinuxMachineDefines LinuxSourceDefines
 
 #define ConnectionFlags		-DUNIXCONN -DTCPCONN
diff -ur mas-cvs-20050207/config/mas.tmpl mas-cvs-mod/config/mas.tmpl
--- mas-cvs-20050207/config/mas.tmpl	2004-06-15 07:35:36.000000000 +1000
+++ mas-cvs-amd64/config/mas.tmpl	2005-02-08 00:22:14.000000000 +1100
@@ -57,6 +57,10 @@
 MAS_ENDIANNESS = MAS_LITTLE_ENDIAN
 #define MAS_WE_ARE_LITTLE_ENDIAN
 
+#elif defined(AMD64Architecture)
+MAS_ENDIANNESS = MAS_LITTLE_ENDIAN
+#define MAS_WE_ARE_LITTLE_ENDIAN
+
 #elif defined(ia64Architecture)
 MAS_ENDIANNESS = MAS_LITTLE_ENDIAN
 #define MAS_WE_ARE_LITTLE_ENDIAN
diff -ur mas-cvs-20050207/config/xfree86.cf mas-cvs-mod/config/xfree86.cf
--- mas-cvs-20050207/config/xfree86.cf	2003-06-27 07:00:26.000000000 +1000
+++ mas-cvs-amd64/config/xfree86.cf	2005-02-07 22:53:17.000000000 +1100
@@ -586,6 +586,12 @@
 #endif
 #endif
 
+#if HasGcc2 && defined(AMD64Architecture)
+# ifndef DefaultGcc2AMD64Opt
+#  define DefaultGcc2AMD64Opt -O3 -fno-strength-reduce
+# endif
+#endif
+
 #ifndef BuildLBXCompatible
 #define BuildLBXCompatible NO
 #endif
diff -ur mas-cvs-20050207/devices/visual/Imakefile mas-cvs-mod/devices/visual/Imakefile
--- mas-cvs-20050207/devices/visual/Imakefile	2003-03-11 09:29:38.000000000 +1100
+++ mas-cvs-amd64/devices/visual/Imakefile	2005-02-09 22:04:49.037874176 +1100
@@ -12,7 +12,7 @@
 XCOMM ################ Architecture dependent ###########################
 
 #ifdef LinuxArchitecture
-ARCHREQUIREDLIBS = -L/usr/X11R6/lib 
+ARCHREQUIREDLIBS = -L/usr/X11R6/LibDirName 
 #endif
 
 #ifdef SunArchitecture
@@ -34,7 +34,7 @@
  LOCAL_LIBRARIES = 
    SYS_LIBRARIES =
          DEPLIBS = 
-    REQUIREDLIBS = $(ARCHREQUIREDLIBS) -L/usr/local/lib -lX11 -lXext -lsrfftw -lsfftw -lm
+    REQUIREDLIBS = $(ARCHREQUIREDLIBS) -L/usr/local/LibDirName -lX11 -lXext -lsrfftw -lsfftw -lm
 
 #include <Library.tmpl>
 
